test/gtest/uct/ib/test_rc.cc

```diff
@@ -312,6 +312,264 @@ UCS_TEST_P(test_rc_flow_control, fc_disabled_flush)
 UCT_RC_INSTANTIATE_TEST_CASE(test_rc_flow_control)
 
 
+#ifdef IBV_HW_TM
+// TODO: Remove when declared in UCT
+#define UCT_RC_MLX5_TAG_BCOPY_MAX     131072
+
+size_t test_rc_mp_xrq::m_rx_counter = 0;
+
+test_rc_mp_xrq::test_rc_mp_xrq() : m_first_received(false),
+                                   m_last_received(false)
+{
+    m_max_hdr = sizeof(ibv_tmh) + sizeof(ibv_rvh);
+    m_uct_comp.count = 512; // We do not need completion func to be invoked
+    m_uct_comp.func  = NULL;
+}
+
+void test_rc_mp_xrq::init()
+{
+    ucs_status_t status1 = uct_config_modify(m_iface_config,
+                                             "RC_TM_MP_NUM_STRIDES", "8");
+    ucs_status_t status2 = uct_config_modify(m_iface_config,
+                                             "RC_TM_ENABLE", "y");
+    uct_test::init();
+
+    if ((status1 != UCS_OK) || (status2 != UCS_OK)) {
```

Comment:
check it before `uct_test::init()` done, since test infrastracture doest't call `cleanup()` when handling an excpetion from `init()`
`test_base::TearDownProxy()` calls `cleanup()` if `m_initialized` set, but `m_initialized` is set after successful `init()` in `test_base::SetUpProxy()`

Reply:
i don't think it should be a problem, init() should cleanup after itself

Reply:
@yosefe when should it clean up after itself?
if we raise an exception, then `uct_test` will be initialized and never cleaned up, isn't?

Reply:
ok, i see the problem. yes, better to check before calling parent init.